### DISCUSSION TITLE
Can't get inline rustanalyzer hints to work

### DISCUSSION THREAD

**Discussion by tschinz** [zed-industries/zed#16670] [Feature Requests]
Is there a method to get inline rustanalyer hints. For example
let x = 0.0;
let x:f64 = 0.0;

where :f64 is inserted by rustanalyzer as it is done on most other IDE's

Thanks

**Comment by SomeoneToIgnore** [comment]
Usually that means the project or Zed are somehow broken, can you share your project and ensure that other editor works with this particular file in that particular project?

Looking at Zed logs and language servers might help to debug this too, there are corresponding commands in Zed command palette.

**Comment by tschinz** [comment]
A cargo new project doesn't even work. It has todo with the Zed or the Settings.
<img width="434" alt="Screenshot 2024-08-22 at 16 35 44" src="https://github.com/user-attachments/assets/ca69fa95-35e5-40d3-a605-330d5ae54176">

My settings are:
```
{
  "base_keymap": "SublimeText",
  "theme": "Ayu Dark",
  "buffer_font_size": 15,
  "buffer_font_family": "Fira Code",
  // Control what info is collected by Zed.
  "telemetry": {
    // Send debug info like crash reports.
    "diagnostics": false,
    // Send anonymized usage data like what languages you're using Zed with.
    "metrics": false
  },
  "journal": {
    // What format to display the hours in
    // May take 2 values:
    // 1. hour12
    // 2. hour24
    "hour_format": "hour24"
  },
  "git": {
    "git_gutter": "tracked_files",
    "inline_blame": {
      "enabled": false
    }
  },
  "show_completion_documentation": true,
  "show_whitespaces": "selection",
  "tab_size": "2",
  "format_on_save": "on",
  "autosave": "on_focus_change"
}

```
And I tried to play with the following without luck:
```
"inlayHints": {
  "enable": true,
  "maxLength": null,
  "lifetimeElisionHints": {
  "useParameterNames": true,
    "enable": "skip_trivial"
  },
  "closureReturnTypeHints": {
    "enable": "always"
  }
}
```

**Comment by SomeoneToIgnore** [comment]
Yes, your settings seem to be quite broken and since inlays are disabled by default, Zed falls back to the defaults.

1. "tab_size": "2", is wrong, there's a tooltip saying that it has to be an integer
2. the whole `inlayHints` section is wrong, starting from the spelling finishing the names themselves, please refer to default settings to see the values: https://github.com/zed-industries/zed/blob/6b9fa68dc521a34a19133d6243a51f82b225c14b/assets/settings/default.json#L286-L301

Fixing the tab size value and adding
```
"inlay_hints": {
    "enabled": true
}
```
seems to be enough to show the hints.

**Reply by tschinz** [reply]
thanks amazing. Sometimes you can't see the forest in midst the trees...
